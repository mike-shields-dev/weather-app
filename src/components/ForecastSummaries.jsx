import React from "react";
import PropTypes from "prop-types";
import styles from "../styles/ForecastSummaries.module.css";
import ForecastSummary from "./ForecastSummary";

const ForecastSummaries = ({
  forecasts,
  handleForecastSelect,
  selectedDate,
}) => {
  return (
    <div
      className={styles["forecast-summaries"]}
      data-testid="forecast-summaries"
    >
      {forecasts.map((forecast) => {
        return (
          <ForecastSummary
            handleForecastSelect={handleForecastSelect}
            key={`forecast-summary-${forecast.date}`}
            selectedDate={selectedDate}
            {...forecast}
          />
        );
      })}
    </div>
  );
};

ForecastSummaries.propTypes = {
  forecasts: PropTypes.arrayOf(
    PropTypes.shape({
      date: PropTypes.number,
      description: PropTypes.string,
      icon: PropTypes.string,
      temperature: PropTypes.shape({
        max: PropTypes.number,
        min: PropTypes.number,
      }),
    })
  ).isRequired,
  handleForecastSelect: PropTypes.func.isRequired,
  selectedDate: PropTypes.number.isRequired,
};

export default ForecastSummaries;
