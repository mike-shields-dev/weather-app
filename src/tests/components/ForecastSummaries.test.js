import React from "react";
import { render } from "@testing-library/react";
import ForecastSummaries from "../../components/ForecastSummaries";

const validProps = [
  {
    date: 1525046400000,
    temperature: {
      max: 11,
      min: 4,
    },
    wind: {
      speed: 13,
      direction: "s",
    },
    humidity: 30,
    description: "Clear",
    icon: "800",
  },
  {
    date: 1525132800000,
    temperature: {
      max: 13,
      min: 8,
    },
    wind: {
      speed: 60,
      direction: "ne",
    },
    humidity: 80,
    description: "Stormy",
    icon: "211",
  },
  {
    date: 1525219200000,
    temperature: {
      max: 1,
      min: -2,
    },
    wind: {
      speed: 5,
      direction: "n",
    },
    humidity: 50,
    description: "Heavy Snow",
    icon: "602",
  },
];

describe("ForecastSummaries", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(
      <ForecastSummaries
        forecasts={validProps}
        handleForecastSelect={jest.fn()}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders forecasts as ForecastSummary components", () => {
    const { getAllByTestId } = render(
      <ForecastSummaries
        forecasts={validProps}
        handleForecastSelect={jest.fn()}
      />
    );
    expect(getAllByTestId("forecast-summary")).toHaveLength(validProps.length);
  });
});
