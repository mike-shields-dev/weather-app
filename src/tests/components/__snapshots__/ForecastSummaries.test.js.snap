// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummaries matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="forecast-summaries"
    data-testid="forecast-summaries"
  >
    <div
      class="forecast-summary shadow"
      data-testid="forecast-summary"
    >
      <h2
        class="forecast-summary__date"
        data-testid="forecast-summary-date"
      >
        <span>
          Mon 30th Apr
        </span>
      </h2>
      <p
        class="forecast-summary__description"
        data-testid="forecast-summary-description"
      >
        Clear
      </p>
      <div
        class="weather-icon-component"
        data-testid="weather-icon-component"
      >
        <i
          class="wi wi-day-sunny"
          name="owm"
          style="transform: scale(2.75); color: rgb(255, 255, 255);"
        />
      </div>
      <p
        class="forecast-summary__temperature"
        data-testid="forecast-summary-temperature"
      >
        <span>
          11
        </span>
        °C
      </p>
      <button
        class="shadow"
        type="button"
      >
        More details...
      </button>
    </div>
    <div
      class="forecast-summary shadow"
      data-testid="forecast-summary"
    >
      <h2
        class="forecast-summary__date"
        data-testid="forecast-summary-date"
      >
        <span>
          Tue 1st May
        </span>
      </h2>
      <p
        class="forecast-summary__description"
        data-testid="forecast-summary-description"
      >
        Stormy
      </p>
      <div
        class="weather-icon-component"
        data-testid="weather-icon-component"
      >
        <i
          class="wi wi-day-thunderstorm"
          name="owm"
          style="transform: scale(2.75); color: rgb(255, 255, 255);"
        />
      </div>
      <p
        class="forecast-summary__temperature"
        data-testid="forecast-summary-temperature"
      >
        <span>
          13
        </span>
        °C
      </p>
      <button
        class="shadow"
        type="button"
      >
        More details...
      </button>
    </div>
    <div
      class="forecast-summary shadow"
      data-testid="forecast-summary"
    >
      <h2
        class="forecast-summary__date"
        data-testid="forecast-summary-date"
      >
        <span>
          Wed 2nd May
        </span>
      </h2>
      <p
        class="forecast-summary__description"
        data-testid="forecast-summary-description"
      >
        Heavy Snow
      </p>
      <div
        class="weather-icon-component"
        data-testid="weather-icon-component"
      >
        <i
          class="wi wi-day-snow"
          name="owm"
          style="transform: scale(2.75); color: rgb(255, 255, 255);"
        />
      </div>
      <p
        class="forecast-summary__temperature"
        data-testid="forecast-summary-temperature"
      >
        <span>
          1
        </span>
        °C
      </p>
      <button
        class="shadow"
        type="button"
      >
        More details...
      </button>
    </div>
  </div>
</DocumentFragment>
`;
